nodestream_plugin_neptune/query_executor.py

```diff
@@ -0,0 +1,160 @@
+        else:
+            partition_size = math.floor(params_count / self.async_partitions)
+
+        """
+        From Dave we found that the sweet spot is around 100 - 200
+        per batch request. Though this is not a hard rule.
+
+        More investigation on performance is needed.
+        """
+        partition_size = 150
+
+        for i in range(0, len(parameters), partition_size):
+            yield {"params": parameters[i : i + partition_size]}
+
+    async def query(self, query_stmt: str, parameters: list):
+        response = None
+        async with self.session.create_client(
+            "neptunedata", region_name=self.region, endpoint_url=self.host
+        ) as client:
+            try:
+                response = await client.execute_open_cypher_query(
+                    openCypherQuery=query_stmt,
+                    # Use json.dumps() to warp dict's key/values in double quotes.
+                    parameters=json.dumps(parameters),
+                )
+
+                code = response["ResponseMetadata"]["HTTPStatusCode"]
+                if code != 200:
+                    self.logger.error(
+                        f"Query `{query_stmt}` failed with response:\n{response}"
+                    )
+
+            except Exception:
+                self.logger.exception(
+                    f"Unexpected error for query: {query_stmt}.", stack_info=True
+                )
+
+        return response
+
+    async def execute(self, query: Query, log_result: bool = False):
+        self.logger.debug(
+            "Executing Cypher Query to Neptune",
+            extra={
+                "query": query.query_statement,
+                "uri": self.host,
+            },
+        )
+        query_stmt = query.query_statement
+
+        # TODO: Replace with proper datatype handling
+        query.parameters["earliest_allowed_time"] = str(
+            query.parameters["earliest_allowed_time"]
+        )
+
+        result = await asyncio.gather(self.query(query_stmt, query.parameters))
+
+        if log_result:
+            for record in result.records:
+                self.logger.info(
+                    "Gathered Query Results",
+                    extra=dict(**record, query=query.query_statement),
+                )
+
+    async def execute_batch(self, query_batch: QueryBatch, log_result: bool = False):
+        query: Query = query_batch.as_query()
+
+        self.logger.debug(
+            "Executing Cypher Query to Neptune",
+            extra={
+                "query": query.query_statement,
+                "uri": self.host,
+            },
+        )
+        query_stmt = query.query_statement
+        requests = (
+            self.query(query_stmt, parameters)
+            for parameters in self._split_parameters(query.parameters["params"])
+        )
+
+        result = await asyncio.gather(*requests)
+
+        if log_result:
+            for record in result.records:
+                self.logger.info(
```

Comment:
Debug or Trace?

Reply:
I think this one is copy-paste from the neo4j code. We put this at info because it included statistics such as the number of relationships and nodes updated. 

We found this information useful when the logger is enabled. I think if the query does not produce the same kind of information downgrading the level is useful. 

There is nuance here that log messages only appear when you either manually configure the logger in a users app or they use the --json flag to enable logging.